### PR TITLE
Feature/mwoar ts 2

### PR DESCRIPTION
**Pull request by ryan-rushton**
Did a couple of more components. Some of the boolean logic here is a little crazy but I am hoping we can streamline it once we give the the containers above a bit of a cleanup too. There doesn't really seem to be an obvious way yet. Splitting out a couple of more smaller components like I did with the Avatar might help?

### CODE REVIEW

File: frontend/src/modules/report/FeatureListContainer/TagViewFeatureList/TagListItem.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  const [expanded, setExpanded] = useState(false);
+  const featureList = tag.features.filter(feature => selectedStatus[feature.calculatedStatus]);
+  const { userName, name, isIgnored } = tag;
+
+  const onAvatarClick = (event: MouseEvent): void => {
+    event.stopPropagation();
+    if (userName && userName !== loggedInUserName) {
+      handleWarningShow(restId, name, loggedInUserName, userName);
+    } else {
+      handleTagAssigned(restId, name, loggedInUserName, userName);
```
Comment (kevinoshea) [inline frontend/src/modules/report/FeatureListContainer/TagViewFeatureList/TagListItem.tsx]: handleTagAssigned has the same condition in it as here. Would it be clearer to put all the logic in one place here so the handlers don't have to make any decisions? eg something like:
if (!username) assignToMe
else if (userName === me) unassign
else showWarning.

File: frontend/src/modules/report/FeatureListContainer/TagViewFeatureList/TagListItem.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    }
+  };
+
+  return isAssignedTagsView && isIgnored ? null : (
+    <>
+      <ListItem button onClick={(): void => setExpanded(!expanded)} className={classes.listItem}>
+        {isEditMode && (
```
Comment (kevinoshea) [inline frontend/src/modules/report/FeatureListContainer/TagViewFeatureList/TagListItem.tsx]: The render methods in some of these components are pretty chunky. Would be good to gradually try to break these up into smaller chunks of renderA(), renderB(), etc.

File: frontend/src/modules/report/FeatureListContainer/TagViewFeatureList/TagListItem.tsx (hunk 1 above)
Comment (ryan-rushton) [reply frontend/src/modules/report/FeatureListContainer/TagViewFeatureList/TagListItem.tsx]: Yeah definitely, this was copy paste so I can clean it up further

File: frontend/src/modules/report/FeatureListContainer/TagViewFeatureList/TagListItem.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  const onAvatarClick = (event: MouseEvent): void => {
+    event.stopPropagation();
+    if (userName && userName !== loggedInUserName) {
+      handleWarningShow(restId, name, loggedInUserName, userName);
+    } else {
+      handleTagAssigned(restId, name, loggedInUserName, userName);
+    }
+  };
+
+  return isAssignedTagsView && isIgnored ? null : (
```
[the remark this answers is not in the document]
Comment (ryan-rushton) [reply frontend/src/modules/report/FeatureListContainer/TagViewFeatureList/TagListItem.tsx]: Yeah agreed, will clean up

File: frontend/src/modules/report/FeatureListContainer/TagViewFeatureList/TagListItem.tsx (hunk 2 above)
Comment (ryan-rushton) [reply frontend/src/modules/report/FeatureListContainer/TagViewFeatureList/TagListItem.tsx]: Lately I have been moving away from render methods to creating more components with more single responsibility like functionality. I agree though this is too big I just didn't want to change too much in this and instead just move the logic over to TS. I will take a pass at it though and create another component or two to split it down further. I am basically looking at it now that if you can pull something out into a render method, you might as well make a small component for it as lets face it, they are just functions but they get all the pro's of being able to memoize stuff if it is rendered alot.

Comment (kevinoshea) [reply frontend/src/modules/report/FeatureListContainer/TagViewFeatureList/TagListItem.tsx]: Yep sounds good.

**Pull request merged**